tests/backup/backup_resiliency_test.go

```diff
@@ -1825,3 +1826,169 @@ var _ = Describe("{CancelAllRunningRestoreJobs}", func() {
+			}
+		})
+
+		Step("Taking backup of applications with incremental iteration of network delay in Milliseconds", func() {
+			log.InfoD("Taking backup of applications with incremental iteration of network delay in Milliseconds")
+			incrementalMilliseconds := 500
+			MinDelayInMilliseconds := 0
+			MaxDelayInMilliseconds := 100000
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				for i := MinDelayInMilliseconds; i <= MaxDelayInMilliseconds; i += incrementalMilliseconds {
+					log.Infof(fmt.Sprintf("Adding a delay of %dms  to the applications nodes", i))
+					err := InjectNetworkErrorInNodes(appNodesMap[namespace], "delay", "add", 0, i)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Adding a delay of %dms to nodes", i))
+					backupName := fmt.Sprintf("%s-%s-delay-%dms-%s", BackupNamePrefix, namespace, i, RandomString(5))
+					appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+					err = CreateBackupWithValidation(ctx, backupName, SourceClusterName, bkpLocationName, backupLocationUID, appContextsToBackup, labelSelectors, BackupOrgID, srcClusterUid, "", "", "", "")
+					if err != nil {
+						log.Infof(fmt.Sprintf("The backup creation failed with error [%v]", err))
+						bkpStatus, bkpReason, err := CheckBackupStatus(backupName, ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("fetching backup status for backup [%s]", backupName))
+						if bkpStatus == api.BackupInfo_StatusInfo_Failed && strings.Contains(bkpReason, "Timed out while waiting for a status update.") {
+							log.InfoD(fmt.Sprintf("Backup [%s] failed when network delay of %dms injected to the applications node", backupName, i))
+							log.Infof(fmt.Sprintf("Deleting the delay of %dms  to the applications nodes", i))
+							err := InjectNetworkErrorInNodes(appNodesMap[namespace], "delay", "del", 0, i)
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Removing delay of %dms from nodes when backup failed", i))
+							log.Infof(fmt.Sprintf("Create the backup when delay of %dms is removed and verify backup is succeded", i))
+							finalBackupName := fmt.Sprintf("%s-%s-nodelay-%s", BackupNamePrefix, namespace, RandomString(5))
+							err = CreateBackupWithValidation(ctx, finalBackupName, SourceClusterName, bkpLocationName, backupLocationUID, appContextsToBackup, labelSelectors, BackupOrgID, srcClusterUid, "", "", "", "")
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s] of namespace (scheduled Context) [%s]", backupName, namespace))
+							backupNamespaceMap[namespace] = finalBackupName
+							break
+						}
+					} else {
+						log.Infof(fmt.Sprintf("Backup [%s] succeded with a network delay of [%d]", backupName, i))
+						log.Infof(fmt.Sprintf("Deleting the delay of %dms  to the applications nodes", i))
+						err := InjectNetworkErrorInNodes(appNodesMap[namespace], "delay", "del", 0, i)
```

Comment:
how about we defer this?

Reply:
in case of failure case InjectNetworkErrorInNodes(appNodesMap[namespace], "delay", "del", 0, i) will be exercised twice if defer it.

Reply:
In case of panic or failure the other tests might be affected as well. I can see InjectNetworkError does not support empty errorInjectionType and refactoring this might be beyond the scope of this PR. 

If everything is in place, it would look something like this. I have not tested it.

```go
defer func() {
    // the function return values should be named to catch this error
    err = Inst().N.InjectNetworkError(nodes, "", "del", 0, 0)
}()
```

Reply:
Currently the test wont cause any panic and Handling empty errorInjectionType wont be ideal case ,since its mandatory to do "del" for current errorInjectionType before injecting any other errorInjectionType, hence it will beyond the scope at testcase level.